utils/staking-miner/src/main.rs

```diff
@@ -298,6 +301,42 @@ enum Solvers {
 	},
 }
 
+#[derive(Debug, Copy, Clone)]
+enum SubmissionStrategy {
+	// Only submit if at the time, we are the best.
+	IfLeading,
+	// Always submit.
+	Always,
+	// Submit if we are leading, or if the solution that's leading is more that the given `Perbill`
+	// better than us. This helps detect obviously fake solutions and still combat them.
+	ClaimBetterThan(Perbill),
+}
+
+/// Custom `impl` to parse `SubmissionStrategy` from CLI.
+///
+/// Possible options:
+/// * --submission-strategy if-leading: only submit if leading
+/// * --submission-strategy always: always submit
+/// * --submission-strategy "percent-better <percent>": submit if submission is `n` percent better. (n > 100 has not effect)
+///
+impl FromStr for SubmissionStrategy {
+	type Err = String;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		let res = if s == "if-leading" {
+			Self::IfLeading
+		} else if s == "always" {
+			Self::Always
+		} else if s.starts_with("percent-better ") {
```

Comment:
this is a little but weird/controversial one need to passing this a string such as:

```$ target/release/staking-miner --uri ws://localhost:9999 --seed //Alice monitor --submission-strategy "percent-better 1" phrag-mms
```

we could just take a number but :point_up: is a bit more readable

Reply:
I think this is fine 👍